golem-worker-executor-base/src/lib.rs

```diff
@@ -100,7 +100,7 @@ const VERSION: &str = golem_version!();
 pub struct RunDetails {
     pub http_port: u16,
     pub grpc_port: u16,
-    pub epoch_thread: std::thread::JoinHandle<()>,
+    pub epoch_thread: std::sync::Mutex<Option<std::thread::JoinHandle<()>>>,
```

Comment:
Can we move the mutex + Option into the cli?
I don't like making the types less precise here as we will always have a joinhandle.

Reply:
- we cannot
- we expose async service creation methods, and i would say, it is preferable to make them returning Send futures in general, which makes them usuable in more situations
- the type is not less precise at all, to actually _join_ a join handle jo have to consume it, which means you have to take it out from the mutex, i would argue that even without the mutex it should have been an option, so one can join it without destroying the whole RunDetails